### PR TITLE
Hermes

### PR DESCRIPTION
**Pull request by dragoslava**
- rscadd: "Add's the Hermes mech to robotics."

Credit to Kyres for sprites

https://forums.aurorastation.org/topic/11276-the-small-ish-robotics-rework/

### CODE REVIEW

**Comment by dragoslava** [comment]
!review

File: code/game/mecha/mecha_construction_paths.dm [hunk 1]
```diff
@@ -1591,3 +1591,216 @@
...
+/////// Hermes
+
+/datum/construction/mecha/hermes_chassis
+	steps = list(list("key"=/obj/item/mecha_parts/part/hermes_torso),//1
+					 list("key"=/obj/item/mecha_parts/part/hermes_head),//2
+					 list("key"=/obj/item/mecha_parts/part/hermes_left_arm),//3
+					 list("key"=/obj/item/mecha_parts/part/hermes_right_arm),//4
+					 list("key"=/obj/item/mecha_parts/part/hermes_left_leg),//5
+					 list("key"=/obj/item/mecha_parts/part/hermes_right_leg)//6
+					)
```
Comment (Karolis2011) [inline code/game/mecha/mecha_construction_paths.dm]: Is there a need for such bizarre indentation?

File: code/game/mecha/mecha_construction_paths.dm [hunk 2]
```diff
@@ -1591,3 +1591,216 @@
...
+
+	custom_action(step, atom/used_atom, mob/user)
```
Comment (Karolis2011) [inline code/game/mecha/mecha_construction_paths.dm]: Please use absolute proc paths
```suggestion
/datum/construction/mecha/hermes_chassis/custom_action(step, atom/used_atom, mob/user)
```

File: code/game/mecha/mecha_construction_paths.dm [hunk 3]
```diff
@@ -1591,3 +1591,216 @@
...
+		user.visible_message("[user] has connected [used_atom] to [holder].", "You connect [used_atom] to [holder]")
+		holder.add_overlay("[used_atom.icon_state]+o")
+		qdel(used_atom)
+		return 1
+
+	action(atom/used_atom,mob/user as mob)
```
Comment (Karolis2011) [inline code/game/mecha/mecha_construction_paths.dm]: Please use absolute proc paths
```suggestion
/datum/construction/mecha/hermes_chassis/action(atom/used_atom,mob/user as mob)
```

File: code/game/mecha/mecha_construction_paths.dm [hunk 4]
```diff
@@ -1591,3 +1591,216 @@
...
+		return check_all_steps(used_atom,user)
+
+	spawn_result()
```
Comment (Karolis2011) [inline code/game/mecha/mecha_construction_paths.dm]: Please use absolute proc paths
```suggestion
/datum/construction/mecha/hermes_chassis/spawn_result()
```

File: code/game/mecha/mecha_construction_paths.dm [hunk 5]
```diff
@@ -1591,3 +1591,216 @@
...
+					 //13
+					 list("key"=/obj/item/weapon/screwdriver,
+					 		"backkey"=/obj/item/weapon/wrench,
+					 		"desc"="The hydraulic systems are connected."),
+					 //14
+					 list("key"=/obj/item/weapon/wrench,
+					 		"desc"="The hydraulic systems are disconnected.")
+					)
+
+	action(atom/used_atom,mob/user as mob)
```
Comment (Karolis2011) [inline code/game/mecha/mecha_construction_paths.dm]:
```suggestion
/datum/construction/reversible/mecha/hermes/action(atom/used_atom,mob/user as mob)
```

File: code/game/mecha/mecha_construction_paths.dm [hunk 6]
```diff
@@ -1591,3 +1591,216 @@
...
+		return check_step(used_atom,user)
+
+	custom_action(index, diff, atom/used_atom, mob/user)
```
Comment (Karolis2011) [inline code/game/mecha/mecha_construction_paths.dm]:
```suggestion
/datum/construction/reversible/mecha/hermes/custom_action(index, diff, atom/used_atom, mob/user)
```

File: code/game/mecha/mecha_construction_paths.dm [hunk 7]
```diff
@@ -1591,3 +1591,216 @@
...
+			if(1)
+				if(diff==FORWARD)
+					user.visible_message("[user] welds external armor layer to [holder].", "You weld external armor layer to [holder].")
+					holder.icon_state = "hermes14"
+				else
+					user.visible_message("[user] unfastens the external armor layer.", "You unfasten the external armor layer.")
+					holder.icon_state = "hermes12"
+		return 1
+
+	spawn_result()
```
Comment (Karolis2011) [inline code/game/mecha/mecha_construction_paths.dm]:
```suggestion
/datum/construction/reversible/mecha/hermes/spawn_result()
```

File: code/game/mecha/mecha_construction_paths.dm [hunk 8]
```diff
@@ -1591,3 +1591,216 @@
...
+		..()
+		feedback_inc("mecha_hermes_created",1)
```
Comment (Karolis2011) [inline code/game/mecha/mecha_construction_paths.dm]: @Arrow768 do we even have feedback database?

Comment (Arrow768) [reply code/game/mecha/mecha_construction_paths.dm]: Yes, but atm we dont really have any stats systems that can be viewed by the public.

File: code/game/mecha/equipment/tools/tools.dm [hunk 9]
```diff
@@ -1,3 +1,5 @@
+
```
Comment (skull132) [inline code/game/mecha/equipment/tools/tools.dm]: wut?

File: code/game/mecha/mecha_construction_paths.dm (hunk 1 above)
Comment (skull132) [reply code/game/mecha/mecha_construction_paths.dm]: It's an attempt to align with tabs, as opposed to spaces.

File: code/game/mecha/mecha_construction_paths.dm [hunk 10]
```diff
@@ -1591,3 +1591,216 @@
...
+	action(atom/used_atom,mob/user as mob)
+		return check_all_steps(used_atom,user)
+
+	spawn_result()
+		var/obj/item/mecha_parts/chassis/const_holder = holder
+		const_holder.construct = new /datum/construction/reversible/mecha/hermes(const_holder)
+		const_holder.icon = 'icons/mecha/mech_construction.dmi'
+		const_holder.icon_state = "hermes0"
+		const_holder.density = 1
+		spawn()
```
Comment (skull132) [inline code/game/mecha/mecha_construction_paths.dm]: Change this to `QDEL_IN`. Look around for examples.

File: code/game/mecha/mecha_parts.dm [hunk 11]
```diff
@@ -299,3 +299,43 @@
...
 	construction_time = 200
 	construction_cost = list(DEFAULT_WALL_MATERIAL=15000)*/
+
+
+//////////// HERMES
+
+/obj/item/mecha_parts/chassis/hermes
+	name = "Hermes Chassis"
+
+	New()
```
Comment (skull132) [inline code/game/mecha/mecha_parts.dm]: `Initialize()`

File: code/game/mecha/mecha_construction_paths.dm [hunk 12]
```diff
@@ -1594,3 +1590,215 @@
...
+					 //13
+					 list("key"=/obj/item/weapon/screwdriver,
+					 		"backkey"=/obj/item/weapon/wrench,
+					 		"desc"="The hydraulic systems are connected."),
+					 //14
+					 list("key"=/obj/item/weapon/wrench,
+					 		"desc"="The hydraulic systems are disconnected.")
+					)
+
+	/datum/construction/reversible/mecha/hermes/action(atom/used_atom,mob/user as mob)
```
Comment (Sindorman) [inline code/game/mecha/mecha_construction_paths.dm]: Why this line and lines all the way until the end of the file have extra tab? I would say just shift+tab lines 1685-1804.

File: code/game/mecha/mecha_wreckage.dm [hunk 13]
```diff
@@ -112,6 +112,25 @@
...
 	icon_state = "seraph-broken"
 
+/obj/effect/decal/mecha_wreckage/hermes
+	name = "Hermes wreckage"
+	icon_state = "hermes-broken"
+
+/obj/effect/decal/mecha_wreckage/hermes/New()
+	..()
+	var/list/parts = list(/obj/item/mecha_parts/part/ripley_torso,
+								/obj/item/mecha_parts/part/ripley_left_arm,
```
Comment (Sindorman) [inline code/game/mecha/mecha_wreckage.dm]: In this case you can use both tabs + spaces to alling it. Try to make it like
```
var/list/parts = list(/obj/item/mecha_parts/part/ripley_torso,
                      /obj/item/mecha_parts/part/ripley_left_arm,
                      /obj/item/mecha_parts/part/ripley_right_arm,
                      /obj/item/mecha_parts/part/ripley_left_leg,
	              /obj/item/mecha_parts/part/ripley_right_leg
                      )
```

**Pull request merged**